### PR TITLE
Created unit tests for `QueryTemplateModel`

### PR DESCRIPTION
**Pull request by pranjal5399**

### CODE REVIEW

File: packages/api/test/helper/constants.ts [hunk 1]
```diff
@@ -19,6 +19,17 @@ const activityLogs = [
     },
 ];
 
+const queryTemplates = [
+    {
+        name: "abc",
```
Comment (itssamuelrowe) [inline packages/api/test/helper/constants.ts:24]: Use values that are more realistic. For example, the name could be "getUsers".

File: packages/api/test/helper/constants.ts [hunk 2]
```diff
@@ -19,6 +19,17 @@ const activityLogs = [
...
+        description: "lorem ipsum dolor amet",
```
Comment (itssamuelrowe) [inline packages/api/test/helper/constants.ts:25]: And the description could be "Gets the list of users who have placed an order."

File: packages/api/test/helper/constants.ts [hunk 3]
```diff
@@ -19,6 +19,17 @@ const activityLogs = [
...
+        resource: "61c9483bcd1cf99c3a211977",
+        app: "61c93a931da4a79d3a109947",
+        content: "abcd",
```
Comment (itssamuelrowe) [inline packages/api/test/helper/constants.ts:28]: And the content could be `SELECT * FROM users as u LEFT JOIN orders as o ON u.id=o.userId`

File: packages/api/test/models/queryTemplate.test.ts [hunk 4]
```diff
@@ -0,0 +1,307 @@
+import { beforeEach } from "mocha";
+import { QueryTemplateModel } from "@hypertool/common";
+import { assert } from "chai";
+import { assertThrowsAsync, queryTemplates } from "../helper";
+
+describe("QueryTemplate Model", function () {
```
Comment (itssamuelrowe) [inline packages/api/test/models/queryTemplate.test.ts:6]: `QueryTemplate model`

The message that you specify to `describe` and the message that you specify to `it` are combined to form a sentence in the output. Keep that in mind.

File: packages/api/test/models/queryTemplate.test.ts [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+            newQuery.name,
+            1,
+            "The name attribute should be of length 1.",
+        );
+    });
+
+    it("should not be created when name is too long", async () => {
+        const newQuery = new QueryTemplateModel({
+            ...queryTemplate,
+            name: new Array(129 + 1).join("a"),
```
Comment (itssamuelrowe) [inline packages/api/test/models/queryTemplate.test.ts:43]: Good!

File: packages/api/test/models/queryTemplate.test.ts [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+
+    it("should not be created when status is invalid", async () => {
+        const newQuery = new QueryTemplateModel({
+            ...queryTemplate,
+            status: "lorem",
+        });
+
+        await assertThrowsAsync(
+            async () => newQuery.save(),
+            "The status attribute is not from defined origins.",
```
Comment (itssamuelrowe) [inline packages/api/test/models/queryTemplate.test.ts:280]: "origins"?

**Review by itssamuelrowe — CHANGES_REQUESTED** [review]

File: packages/api/test/helper/constants.ts [hunk 7]
```diff
@@ -29,4 +40,42 @@ const appData = {
...
...QpNO2P73opDWAbBBDbQceEeKkMt2jfXxXZR61SD1QdpA0SrhRBJkqQgQheL/4h0yLik47S8T1Xslas1olzgB1WA7e4d+00S+g0vdTeHw0Eu1EHQa7Zll5Z2N/Bea9onZ697gbKvy8OWAwmuWaGQRuDoR4ha+0uJA1XE9Gz9u1y3RVT7cclcPcCo9VoUxRR1aCb+y6K0bQJKWsyFIo32ys+zlgDUdVcO7SEg/znRvoJPonNti4hrRJJAA5krRXFoKNFtJup+Z7ubjv+A8lp4s913+mXm38c8/dVGfUnmU9tcDdRKz4UKtXPBdXXDxJxO80MdPqEzD2yyeXFQntJY49J9NVKwqtBg7GCFKeemksMSezR/eHPiFeW9ZrxLTKz1NgOq6sY5plhhJVbGjCcFV22JDZ4g81ZscCJBkc1ZU5KlQgRCEIEo0c2p2+qlNaAICUCE5QkgSoQga4qBVMuUyoYBUNokqKtE5jYCbWqhoJcYASvcGiSYA4rO4vfF4hvyj38Ut4Zz2oeIXhqOLjt+6OQXO1uCyoCDAdAPgVFknYKZ+yGGkjks5/bX19LbFMOo1Wj4jA7lUgBw8HDVZi5w59A6SWcHcvFbO1thAB2hdv2FnKfFTrxzU9mPNrF/wAYRl0lNaSqypWh7gNg5wHqVItnyVxe3oc/a1two1eJVhbN0XS3tWhr6ztAwENmNXcDHGPr4K2M3V4z3qYnaZhNSnSdmeCXbBoHyzu4zx4LviVy14JYZG/X0VEK+u8+K7tpujOdOX5rrzPjORxat1rtQLifRQm0ySrN4Dt9DwPA+PJNY3KdRH3upU+nNlHQjmCoOHyMvSQrJ7wCFCayHvH80+uqJjQWynNcQNpUKxGis20+6rIRSGu2T6b6lMy06cuHooxIa4knZd7WvmbLuZEpEWLi1xFj9Hd13I7HwKmrPuocQnW9++mYfq33HgVKLF6hQP8AN6XM+iRSheIQhQBIUIQcbj5VGo/MEIUVMGM/7R8QszU+UoQq6+2mDrbdWb9vMfRKhRCri2+UeAXcoQtGTxu5+d39bvqVOw/dCF572J9NHbfKnYx/sUvF/wD7FCFr4vuub+R9RQW3zjxH1V1c/KkQt3Mqaic/5G/3fVCFMRpGO4TH/wC4fAIQiIv8P2VvT28kqFZChvPmKlUPkH3xQhRBYUuPkot9shClFQkIQiH/2Q==",
```
Comment (itssamuelrowe) [inline packages/api/test/helper/constants.ts:52]: What is this? You can use a smaller link like: https://i.picsum.photos/id/496/400/400.jpg?hmac=8or0mw_rugRSBRDFhLAj770EHslVmfHZEu0a_NKkxoY

File: packages/api/test/models/queryTemplate.test.ts [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+        );
+    });
+
+    it("should not be created when status is undefined", async () => {
```
Comment (itssamuelrowe) [inline packages/api/test/models/queryTemplate.test.ts:284]: `QueryTemplate` can be created without `status`. You can remove this test case.

File: packages/common/test/models/resource.test.ts [hunk 9]
```diff
@@ -0,0 +1,649 @@
...
+            ...resource,
+            name: new Array(257 + 1).join("a"),
+        });
+        await assertThrowsAsync(
+            async () => newResource.save(),
+            "The name attribute should have a maximum of 256 characters.",
+        );
+    });
+
+    it("should not be created when name is undefined", async () => {
```
Comment (itssamuelrowe) [inline packages/common/test/models/resource.test.ts:51]: Update `ResourceModel` to add `required: true` to `name` attribute.

```
        name: {
            type: String,
            minlength: 0,
            maxlength: 256,
            trim: true,
            required: true
        },     
```

File: packages/common/test/models/resource.test.ts [hunk 10]
```diff
@@ -0,0 +1,649 @@
...
+            name: undefined,
+        });
+
+        await assertThrowsAsync(
+            async () => newResource.save(),
+            "The name attribute is required.",
+        );
+    });
+
+    it("should not be created when name is null", async () => {
```
Comment (itssamuelrowe) [inline packages/common/test/models/resource.test.ts:63]: If you add required to `name` attribute, this test case will also pass.

File: packages/api/test/models/user.test.ts [hunk 11]
```diff
@@ -0,0 +1,564 @@
...
+            gender: "lorem",
+        });
+
+        await assertThrowsAsync(
+            async () => newUser.save(),
+            "The gender attribute is not from defined options.",
+        );
+    });
+
+    it("should not be created when gender is undefined", async () => {
```
Comment (itssamuelrowe) [inline packages/api/test/models/user.test.ts:237]: It seems like you haven't deleted the files from the API package. :/

**Review by itssamuelrowe — CHANGES_REQUESTED** [review]

**Review by joelerego — APPROVED** [review]

**Review by itssamuelrowe — APPROVED** [review]

**Pull request merged**